vllm/v1/engine/processor.py

```diff
@@ -278,10 +280,21 @@ def process_inputs(
             else:
                 sorted_mm_inputs = individual_mm_inputs
 
+        # Generate prompt kv block hashes
+        prompt_kv_block_hashes = hash_request_tokens(
+            get_hash_fn_by_name(self.cache_config.prefix_caching_hash_algo),
```

Comment:
Just make the hash_fn an instance field?

Reply:
You meant like `self.cache_config.prefix_caching_hash_fn`? I was thinking about this too but not sure if there would be any issue transferring the function around and through zmq/msgpack (likely not?) Will try then.

Reply:
Oh sorry I just meant you could have this in the Processor constructor instead:
```python
self.hash_fn = get_hash_fn_by_name(self.cache_config.prefix_caching_hash_algo)
```

Agree that it might be better to not have the function in the config itself.

Reply:
Ah ok make sense. Revert and make it a Processor attribute. Also rename `self.use_hash` to be `self.hash_mm_inputs` to differentiate them.